src/pykeen/datasets/inductive/base.py

```diff
@@ -0,0 +1,316 @@
+        """
+        if cache_root is None:
+            cache_root = PYKEEN_DATASETS
+        cache_root = pathlib.Path(cache_root).resolve()
+        cache_root = self._extend_cache_root(cache_root=cache_root)
+        # add v1 / v2 / v3 / v4 for inductive splits if available
+        if version is not None:
+            cache_root = cache_root / version
+        cache_root.mkdir(parents=True, exist_ok=True)
+        if sep_train_inference:
+            # generate subfolders 'training' and  'inference'
+            training = cache_root / "training"
+            training.mkdir(parents=True, exist_ok=True)
+            inference = cache_root / "inference"
+            inference.mkdir(parents=True, exist_ok=True)
+        logger.debug("using cache root at %s", cache_root.as_uri())
+        return cache_root
+
+    def _extend_cache_root(self, cache_root: pathlib.Path) -> pathlib.Path:
+        """Get appropriate cache sub-directory."""
+        return cache_root.joinpath(self.__class__.__name__.lower())
+
+
+class DisjointInductivePathDataset(LazyInductiveDataset):
+    """Contains a lazy reference to a training, inductive inference, inductive testing, and inductive validation dataset.
+    In this dataset, inductive inference is disjoint with the transductive train
+    """
+
+    def __init__(
+        self,
+        transductive_training_path: Union[str, pathlib.Path],
+        inductive_inference_path: Union[str, pathlib.Path],
+        inductive_testing_path: Union[str, pathlib.Path],
+        inductive_validation_path: Union[str, str, pathlib.Path],
+        eager: bool = False,
+        create_inverse_triples: bool = False,
+        load_triples_kwargs: Optional[Mapping[str, Any]] = None,
+    ) -> None:
+        """Initialize the dataset.
+
+        :param transductive_training_path: Path to the training triples file or training triples file.
+        :param inductive_inference_path: Path to the inductive inference triples file or training triples file.
+        :param inductive_testing_path: Path to the testing triples file or testing triples file.
+        :param inductive_validation_path: Path to the validation triples file or validation triples file.
+        :param eager: Should the data be loaded eagerly? Defaults to false.
+        :param create_inverse_triples: Should inverse triples be created? Defaults to false.
+        :param load_triples_kwargs: Arguments to pass through to :func:`TriplesFactory.from_path`
+            and ultimately through to :func:`pykeen.triples.utils.load_triples`.
+        """
+        self.transductive_training_path = pathlib.Path(transductive_training_path)
+        self.inductive_inference_path = pathlib.Path(inductive_inference_path)
+        self.inductive_testing_path = pathlib.Path(inductive_testing_path)
+        self.inductive_validation_path = pathlib.Path(inductive_validation_path)
+
+        self.create_inverse_triples = create_inverse_triples
```

Comment:
not really related to this PR, but this should really be removed from the triples factory, and transparently encapsulated in the model